lib/rmt/cli/mirror.rb

```diff
@@ -1,46 +1,55 @@
 # rubocop:disable Rails/Output
+class RMT::CLI::Mirror < RMT::CLI::Base
 
-module RMT::CLI::Mirror
+  default_task :repos
 
-  def self.mirror(repository_url = nil, local_path = nil)
-    require 'rmt/mirror'
-    require 'rmt/config'
+  desc 'repos', 'Mirror enabled repositories'
+  def repos
+    RMT::CLI::Base.handle_exceptions { mirror }
+  end
 
-    if repository_url
-      local_path ||= Repository.make_local_path(repository_url)
-      mirror_one_repo(repository_url, local_path)
-      return
-    end
+  desc 'custom URL', 'Mirror a custom repository URL'
+  def custom(url, path = nil)
+    url += '/' unless url.end_with?('/')
+    path ||= Repository.make_local_path(url)
+    mirror_one_repo(url, path)
+  end
 
-    repositories = Repository.where(mirroring_enabled: true)
+  no_commands do
+    def mirror(base_dir: nil, from_dir: nil, repos: Repository.where(mirroring_enabled: true))
+      require 'rmt/mirror'
+      require 'rmt/config'
 
-    if repositories.empty?
-      warn 'There are no repositories marked for mirroring.'
-      return
-    end
+      base_dir ||= RMT::DEFAULT_MIRROR_DIR
 
-    repositories.each do |repository|
-      begin
-        puts "Mirroring repository #{repository.name}"
-        mirror_one_repo(repository.external_url, repository.local_path, repository.auth_token)
-        repository.refresh_timestamp!
-      rescue RMT::Mirror::Exception => e
-        warn e.to_s
+      if repos.empty?
+        warn 'There are no repositories marked for mirroring.'
+        return
+      end
+
+      repos.each do |repository|
+        begin
+          puts "Mirroring repository #{repository.name} from #{from_dir || 'SCC'} to #{base_dir}"
+          repo_url = repository.external_url
+          repo_url.sub!(/.*(?=SUSE)/, "file://#{from_dir}/") if from_dir
```

Comment:
Hmm, we have some repos like Nvidia and such, I'm not sure if the URL always contains `SUSE` string.